pkg/orchestrator/provider/kubernetes/provider.go

```diff
@@ -113,10 +120,215 @@ func (p *Provider) DiscoverAssets(ctx context.Context) provider.AssetDiscoverer
+	for _, discoverer := range discoverers.Items {
+		discovererEndpoint := net.JoinHostPort(discoverer.Status.PodIP, "8080")
+		clients = append(clients, containerruntimediscovery.NewClient(discovererEndpoint))
+	}
+
+	switch value := objectType.(type) {
+	case models.ContainerImageInfo:
+		var pickedClient *containerruntimediscovery.Client
+		for _, client := range clients {
+			_, err := client.GetImage(ctx, value.ImageID)
+			if err == nil {
+				pickedClient = client
+				break
+			}
+		}
+		if pickedClient == nil {
+			return fmt.Errorf("unable to find image ID %s in any discoverer", value.ImageID)
+		}
+
+		err := p.runScannerJob(ctx, config, pickedClient.ExportImageURL(ctx, value.ImageID))
+		if err != nil {
+			// TODO(sambetts) Make runScannerJob idempotent and
+			// change this to a normal Errorf.
+			return provider.FatalErrorf("unable to run scanner job: %w", err)
+		}
+	case models.ContainerInfo:
+		var pickedClient *containerruntimediscovery.Client
+		for _, client := range clients {
+			_, err := client.GetContainer(ctx, value.ContainerID)
+			if err == nil {
+				pickedClient = client
+				break
+			}
+		}
+		if pickedClient == nil {
+			return fmt.Errorf("unable to find container ID %s in any discoverer", value.ContainerID)
+		}
+
+		err := p.runScannerJob(ctx, config, pickedClient.ExportContainerURL(ctx, value.ContainerID))
+		if err != nil {
+			// TODO(sambetts) Make runScannerJob idempotent and
+			// change this to a normal Errorf.
+			return provider.FatalErrorf("unable to run scanner job: %w", err)
+		}
+	default:
+		return provider.FatalErrorf("failed to scan asset object type %T: Not implemented", value)
+	}
+
+	return nil
+}
+
+// mountPointPath defines the location in the container where assets will be mounted.
+var (
+	mountPointPath  = "/mnt/snapshot"
+	archiveLocation = mountPointPath + "/image.tar"
+)
+
+func (p *Provider) generateScanConfig(config *provider.ScanJobConfig) ([]byte, error) {
+	// Add volume mount point to family configuration
+	familiesConfig := families.Config{}
+	err := yaml.Unmarshal([]byte(config.ScannerCLIConfig), &familiesConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal family scan configuration: %w", err)
+	}
+
+	families.SetOciArchiveForFamiliesInput([]string{archiveLocation}, &familiesConfig)
+	familiesConfigByte, err := yaml.Marshal(familiesConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal family scan configuration: %w", err)
+	}
+
+	return familiesConfigByte, nil
 }
 
-func (p *Provider) RemoveAssetScan(context.Context, *provider.ScanJobConfig) error {
-	return fmt.Errorf("not implemented")
+func (p *Provider) runScannerJob(ctx context.Context, config *provider.ScanJobConfig, sourceURL string) error {
+	configBytes, err := p.generateScanConfig(config)
+	if err != nil {
+		return fmt.Errorf("failed to generate scanner config yaml: %w", err)
+	}
+
+	jobName := fmt.Sprintf("vmclarity-scan-%s", config.AssetScanID)
```

Comment:
`scanName` seems a bit better because its the name for both the pod, container, and configmap

Reply:
I believe it is called `jobName` because it is a batch job object. So I think the naming is correct in this case.